rasa_nlu/data_router.py

```diff
@@ -336,6 +343,28 @@ def training_errback(failure):
         self._current_training_processes += 1
         self.project_store[project].current_training_processes += 1
 
+        # tensorflow training is not executed in a separate thread, as this may
+        # cause training to freeze
+        if self._tf_in_pipeline(train_config):
+            try:
+                model_path = do_train_in_worker(
+                    train_config,
+                    data_file,
+                    path=self.project_dir,
+                    project=project,
+                    fixed_model_name=model_name,
+                    storage=self.remote_storage)
+                model_dir = os.path.basename(os.path.normpath(model_path))
+                self.project_store[project].update(model_dir)
+                return model_dir
+            except TrainingException as e:
+                logger.warn(e)
+                target_project = self.project_store.get(
+                    e.failed_target_project)
+                if target_project:
+                    target_project.status = 0
+                raise e
+
```

Comment:
Please add an Else here, makes it easier to understand that the second part is only run otherwise